File: iniciante/python/1047-tempo-de-jogo-com-minutos.py
def entrada():
    tempo = list(map(int, input().split()))
    hora_inicial = tempo[0]
    minuto_inicial = tempo[1]
    hora_final = tempo[2]
    minuto_final = tempo[3]
    return hora_inicial, minuto_inicial, hora_final, minuto_final


def tempo_de_jogo():
    hora_i, minuto_i, hora_f, minuto_f = entrada()
    minutos_totais_jogado = (hora_f - hora_i) * 60 + minuto_f - minuto_i
    hora_jogada = 0
    minuto_jogado = 0
    if minutos_totais_jogado > 0:
        hora_jogada = minutos_totais_jogado // 60
        minuto_jogado = minutos_totais_jogado % 60
    elif minutos_totais_jogado < 0:
        minutos_totais_jogado = 24 * 60 + minutos_totais_jogado
        hora_jogada = minutos_totais_jogado // 60
        minuto_jogado = minutos_totais_jogado % 60
    else:
        hora_jogada = 24
        minuto_jogado = 0
    print(f'O JOGO DUROU {hora_jogada} HORA(S) E {minuto_jogado} MINUTO(S)')


tempo_de_jogo()

File: iniciante/c/1047-tempo-de-jogo-com-minutos.c
#include <stdio.h>

int main(){
    int hora_inicial, minuto_inicial, hora_final, minuto_final, hora_jogada, minuto_jogado, minuto_total;

    // Entrada do usuário
    scanf("%d", &hora_inicial);
    scanf("%d", &minuto_inicial);
    scanf("%d", &hora_final);
    scanf("%d", &minuto_final);

    // Imprime o tempo que durou o jogo
    if (hora_inicial == hora_final && minuto_inicial == minuto_final){
        hora_jogada = 24;
        minuto_jogado = 0;
    }
    else{
        minuto_total = hora_final * 60 + minuto_final - (hora_inicial * 60 + minuto_inicial);
        if (minuto_total > 0){
            hora_jogada = minuto_total / 60;
            minuto_jogado = minuto_total % 60;
        }
        else{
            minuto_total+= 24 * 60;
            hora_jogada = minuto_total / 60;
            minuto_jogado = minuto_total % 60;
        }
    }
    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", hora_jogada, minuto_jogado);

    return 0;
}
